File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc157/ABC157_B.java
package com.fengtian.qianer.atcoder.abc.abc157;

import java.util.Scanner;

public class ABC157_B {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int[][] A = new int[3][3];
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                A[i][j] = sc.nextInt();
            }
        }

        final int N = sc.nextInt();
        int[] b = new int[N];
        for (int i = 0; i < N; i++) {
            b[i] = sc.nextInt();
        }

        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                for (int k = 0; k < N; k++) {
                    if (A[i][j] == b[k]) {
                        A[i][j] = 0;
                    }
                }
            }
        }

        if (A[0][0] + A[0][1] + A[0][2] == 0
        || A[1][0] + A[1][1] + A[1][2] == 0
        || A[2][0] + A[2][1] + A[2][2] == 0
        || A[0][0] + A[1][0] + A[2][0] == 0
        || A[0][1] + A[1][1] + A[2][1] == 0
        || A[0][2] + A[1][2] + A[2][2] == 0
        || A[0][0] + A[1][1] + A[2][2] == 0
        || A[0][2] + A[1][1] + A[2][0] == 0) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }
}

File: kotlin/src/abc/abc157/ABC157_B.kt
package abc.abc157

import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    val a = Array(3) { Array(3) { it -> sc.nextInt() } }

    val n = sc.nextInt()

    for (i in 0 until n) {
        val b = sc.nextInt()
        for (j in 0 until 3) {
            for (k in 0 until 3) {
                if (a[j][k] == b) a[j][k] = 0
            }
        }
    }

    var bingo = false
    if (a[0][0] == a[0][1] && a[0][1] == a[0][2]) bingo = true
    if (a[1][0] == a[1][1] && a[1][1] == a[1][2]) bingo = true
    if (a[2][0] == a[2][1] && a[2][1] == a[2][2]) bingo = true
    if (a[0][0] == a[1][0] && a[1][0] == a[2][0]) bingo = true
    if (a[0][1] == a[1][1] && a[1][1] == a[2][1]) bingo = true
    if (a[0][2] == a[1][2] && a[1][2] == a[2][2]) bingo = true
    if (a[0][0] == a[1][1] && a[1][1] == a[2][2]) bingo = true
    if (a[2][0] == a[1][1] && a[1][1] == a[0][2]) bingo = true

    if (bingo) {
        println("Yes")
    } else {
        println("No")
    }
}